File: src/usv_tf/scripts/world_tf_broadcaster.py
#!/usr/bin/env python  
import roslib
import rospy
import tf
import sys
from nav_msgs.msg import Odometry

def handle_turtle_pose(msg, br):
    # r = rospy.Rate(102)
    br.sendTransform((msg.pose.pose.position.x, msg.pose.pose.position.y, msg.pose.pose.position.z), (msg.pose.pose.orientation.x, msg.pose.pose.orientation.y, msg.pose.pose.orientation.z, msg.pose.pose.orientation.w), rospy.Time.now(), sys.argv[1] + "_tf/base_link", sys.argv[1] + "_tf/odom")
    # r.sleep()

if __name__ == '__main__':
    rospy.init_node(sys.argv[1] + "world_tf_broadcaster")
    br = tf.TransformBroadcaster()
    rospy.Subscriber("state", Odometry, handle_turtle_pose, br)
    rospy.spin()

File: src/usv_tf/src/world_tf_broadcaster.cpp
#include <ros/ros.h>
#include <tf/transform_broadcaster.h>
#include <nav_msgs/Odometry.h>

void getTransform(const nav_msgs::Odometry::ConstPtr& msg){
    tf::TransformBroadcaster broadcaster;
    broadcaster.sendTransform(
        tf::StampedTransform(
        tf::Transform(tf::Quaternion(msg->pose.pose.orientation.x, msg->pose.pose.orientation.y, msg->pose.pose.orientation.z, msg->pose.pose.orientation.w), tf::Vector3(msg->pose.pose.position.x, msg->pose.pose.position.y, msg->pose.pose.position.z)),
        ros::Time::now(), "world", "base_link"));
}

int main(int argc, char** argv){
    ros::init(argc, argv, "world_tf_publisher");
    ros::NodeHandle n;

    ros::Rate r(100);
    ros::Subscriber sub = n.subscribe("state", 100, getTransform);

    while(n.ok()){
        ros::spinOnce();
        r.sleep();
    }
}
